### ISSUE TITLE
Querying ALB logs with partition

### ISSUE THREAD

**Issue by fubar** [awsdocs/amazon-athena-user-guide#77]
The guide at https://github.com/awsdocs/amazon-athena-user-guide/blob/main/doc_source/application-load-balancer-logs.md refers to a 3rd party solution in order to partition a table for ALB logs, which involves data transformation via a custom Lambda function. However, it is possible to add partitioning to the schema in the same manner as the "Querying AWS WAF Logs" guide at https://github.com/awsdocs/amazon-athena-user-guide/blob/main/doc_source/waf-logs.md describes:

```sql
CREATE EXTERNAL TABLE IF NOT EXISTS alb_logs (
  ...
)

PARTITIONED BY
(
 day STRING
)

ROW FORMAT SERDE 'org.apache.hadoop.hive.serde2.RegexSerDe'
WITH SERDEPROPERTIES (
  ...
)
LOCATION 's3://your-alb-logs-directory/AWSLogs/<ACCOUNT-ID>/elasticloadbalancing/<REGION>/'

TBLPROPERTIES
(
 "projection.enabled" = "true",
 "projection.day.type" = "date",
 "projection.day.range" = "2022/01/01,NOW",
 "projection.day.format" = "yyyy/MM/dd",
 "projection.day.interval" = "1",
 "projection.day.interval.unit" = "DAYS",
 "storage.location.template" = "s3://your-alb-logs-directory/AWSLogs/<ACCOUNT-ID>/elasticloadbalancing/<REGION>/${day}"
)
```

In this query, the `PARTITIONED BY` and `TBLPROPERTIES` statements are added to the "create table" query from the guide, the rest is the same. Querying the table with e.g. `select * from alb_logs where day = '2022/04/08'` then works as expected.

Is there a reason this is not mentioned in the guide? Why does it refer to a more complicated solution that involves a Lambda function?

Thanks!

**Comment by taammann** [comment]
The link to the older solution dates back to a time before partition projection was available in Athena. We'll get the ALB page updated accordingly. Thanks for the catch!

**Comment by fubar** [comment]
I did notice that files can contain entries that are older than the file's date-based path would indicate. For example, the first file written in `2022/04/08` may contain records from the previous day. In our case, I noticed records from a few minutes before midnight; ie. the system does not start new files when the date boundary rolls over. Instead, I assume that the current date is used whenever a file is finalized.

In other words, running a query like `where day = '2022/04/08'` can include records from 2022/04/07. This is still useful to us because it prevents querying the entire dataset, but it's an important thing to know and it would be worth pointing out in the docs as well.

Thanks!

**Comment by taammann** [comment]
Interesting. I suspect that has to do with the fact that a log file for each load balancer node is published every 5 minutes (see [Access log files](https://docs.aws.amazon.com/elasticloadbalancing/latest/application/load-balancer-access-logs.html#access-log-file-format)), which means a log file's contents could span a date boundary. Since that behavior has more to do with ALB than Athena, I'd suggest posting your suggestion on the [Access logs for your Application Load Balancer](https://docs.aws.amazon.com/elasticloadbalancing/latest/application/load-balancer-access-logs.html) page.
